src/app/core/guards/auth.guard.ts

```diff
@@ -13,14 +16,29 @@ import { User } from '../interface/user';
 @Injectable({
   providedIn: 'root',
 })
-export class AuthGuard implements CanActivate {
+export class AuthGuard implements CanActivate, CanLoad {
   private user!: User | null;
 
   constructor(private authService: AuthService, private router: Router) {
     this.authService.authenticatedUser.subscribe((response) => {
       this.user = response;
     });
   }
+  canLoad(
+    route: Route,
+    segments: UrlSegment[]
+  ):
+    | boolean
+    | UrlTree
+    | Observable<boolean | UrlTree>
+    | Promise<boolean | UrlTree> {
+    if (this.user) {
+      return true;
+    } else {
+      this.router.navigate(['/login']);
+      return false;
+    }
```

Comment:
for reusability purpose, we can create a function and call the function in both canLoad and canActivate interface

Reply:
@ban-wtag apu resolved it.